### ISSUE TITLE
New chapters don't show on MTP

### ISSUE THREAD

**Issue by Beanow** [raulhaag/MiMangaNu#202]
Due to a caching bug that's been in android since 2012.
See: http://stackoverflow.com/questions/13737261/nexus-4-not-showing-files-via-mtp

Can you try if the following helps?
```java
MediaScannerConnection.scanFile(this, new String[] { file.getAbsolutePath() }, null, null);
```

**Comment by Johndeep** [comment]
To be honest, I know this and always do some rescan with **SD Scanner** before interacting with the pc by mtp. But didn't know this is actually a bug though..

**Comment by Beanow** [comment]
@Johndeep I've tried different SD scanners, it just made a 4kb file where the folders should be. I ended up transfering the files over samba instead.

**Comment by raulhaag** [comment]
it's a system problem if i make 
MediaScannerConnection.scanFile(this, new String[] { file.getAbsolutePath() }, null, null);
for every downloaded image it go to be realy slow

**Comment by Beanow** [comment]
Certainly running the scan doesn't take as long as downloading the image. So if it's done in parallel with the next image to be downloaded it should be fine right?

**Comment by Johndeep** [comment]
But why not consulate the google team? If it's a system problem, and even if we always make the mediascan (which seems to be slow), this won't solve the other apps problem, right?

If the media scan you said creates only 4kb files, where the folders should be, then it could be other reasons, like corrupted sdcard, not in sync with mtp and OS or even worse.. I think. If that's the case, this simple media scan function won't do any better, I afraid.

**Comment by Beanow** [comment]
@Johndeep have a read of this issue reported to google in 2012 if you want to know about that.
https://code.google.com/p/android/issues/detail?id=38282

**Comment by Johndeep** [comment]
Wow, what a healthy conversation over there (even with a continuation at https://code.google.com/p/android/issues/detail?id=195362), didn't know this problem broke out like that.. just like a volcano, which is still active..

But heck! Such a troublesome bug.. if there is time, I'll looking into it, if nobody else does.
So is there a efficient way to implement this without slowing everything down..

**Comment by Beanow** [comment]
@Johndeep well my hunch would be to use a thread for downloading and a thread for processing. When 1 image is downloaded, let the media scan run over that one file in a second thread. So the scanning will not block the next image download.

But as with everything performance, you won't really know until you've tested it.

**Comment by Johndeep** [comment]
Does this do any difference, if you activate the "hide gallery" option by the way? (which creates the .nomedia file in the MiMangaNu folder).
Because I can see this affects the entire device media finding on the android device, but I don't really know if this also affects the mtp scanning.

**Comment by Johndeep** [comment]
Okay, the media scan and mtp is altogether totally bugged and screws everything up. On the one hand, if I only scan (with a third party app) specific folders, mtp only shows these folders, all the others are non-existent. But if I make a whole scan (again, with another third party app, this time from fdroid), all the datas shows up in mtp, **but** several apps, like vanilla music or the gallery won't find any music or images anymore..

I somehow feel the problem of @Beanow, sorry for not taking this serious. (Even if this still not affecting MiMangaNu, **but** a bunch of other medias..)

Maybe I have a still bugged cyanogenmod (12.1), or the media-scan is altogether wasted resources.
Like you suggested, samba is one way, if anyone has UMS (USB Mass storage), then use that.

**Comment by raulhaag** [comment]
non app issue closed
